### PR TITLE
 Core: Zero-config Typescript e2e test

### PR DESCRIPTION
**Pull request by tooppaaa**
Relates to #10813 but can be merged separately

### CODE REVIEW

**Review by mrmckeb — APPROVED** [review]

File: scripts/run-e2e.ts [hunk 1]
```diff
@@ -134,6 +141,28 @@ const addRequiredDeps = async ({ cwd, additionalDeps }: Options) => {
...
+        jsx: 'preserve',
+        skipLibCheck: true,
+        strict: true,
+      },
+      include: ['src/*'],
+    };
+    const tsConfigJsonPath = path.resolve(cwd, 'tsconfig.json');
+    await writeJSON(tsConfigJsonPath, tsConfig, { encoding: 'utf8', spaces: 2 });
+  } catch (e) {
+    logger.error(`🚨 Creating tsconfig.json failed`);
```
Comment (shilman) [inline scripts/run-e2e.ts:161]: Love these emojis 😍

**Review by shilman — APPROVED** [review]
LGTM!

File: scripts/run-e2e.ts (hunk 1 above)
Comment (tooppaaa) [reply scripts/run-e2e.ts:161]: That's what I can see on windows 🥴
image

File: scripts/run-e2e.ts [hunk 2]
```diff
@@ -134,6 +141,28 @@ const addRequiredDeps = async ({ cwd, additionalDeps }: Options) => {
   }
 };
 
+const addTypescript = async ({ cwd }: Options) => {
```
Comment (yannbf) [inline scripts/run-e2e.ts:144]: Maybe it's a good idea to pass a typescript version and use latest as default? To allow more flexibility for testing different scenarios

Comment (tooppaaa) [reply scripts/run-e2e.ts:144]: I thought about this as well but to keep things simple for now I think it's best to omit this for now

**Comment by ndelangen** [comment]
@tooppaaa is this ready?

**Comment by tooppaaa** [comment]
Some unexpected changes and examples-v2 are red :/ I guess not !

**Pull request merged**